[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>,cmp).

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71298
* #71186



Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

Commit:
[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>).

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

Commit:
Update on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>)."


Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

Commit:
rebase on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>)."


Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

Commit:
Update on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>)."


Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

navahgar commented on torch/csrc/jit/tensorexpr/lowerings.cpp:
```diff
@@ -80,22 +80,114 @@ int nnc_lowerings_lazy_registration() {
+  DEFINE_COMPARISON_SCALAR_OP_LOWERING(ge, cast<bool>(a >= b))
+#undef DEFINE_COMPARISON_SCALAR_OP_LOWERING
+
+#define DEFINE_INT_SCALAR_OP_LOWERING(op_name, op)                        \
```
nit: s/DEFINE_INT_SCALAR_OP_LOWERING/DEFINE_BITWISE_SCALAR_OP_LOWERING

navahgar commented on torch/csrc/jit/tensorexpr/lowerings.cpp:
```diff
@@ -80,22 +80,114 @@ int nnc_lowerings_lazy_registration() {
+  DEFINE_INT_SCALAR_OP_LOWERING(__rshift__, a >> b)
+#undef DEFINE_INT_SCALAR_OP_LOWERING
+
+#define DEFINE_BOOL_SCALAR_OP_LOWERING(op_name, op)                       \
```
nit: s/DEFINE_BOOL_SCALAR_OP_LOWERING/DEFINE_LOGICAL_SCALAR_OP_LOWERING

navahgar commented on test/test_jit_fuser_te.py:
```diff
@@ -1386,6 +1386,62 @@ def apply(fn):
+                "aten::div",
+                "aten::lt",
+                "aten::le",
+                "aten::eq",
```
`aten::ne` missing

navahgar reviewed (approved):
LGTM.

Commit:
Update on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>,cmp)."




Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

Commit:
lints on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>,cmp)."




Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).

Commit:
Update on "[TensorExpr] Add lowerings for scalar binary ops (+,-,*,/,&,|,^,<<,>>,cmp)."




Differential Revision: [D33576534](https://our.internmc.facebook.com/intern/diff/D33576534)

[ghstack-poisoned]

ZolotukhinM commented:
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33576534).
